### PR TITLE
Substantial rewrite of the Cookbook guide

### LINKED ISSUE #433: Cookbook guide is inconsistent with latest template
The section [Update repository-specific text](https://projectpythia.org/cookbook-guide.html#update-repository-specific-text) currently says that "tags" and "thumbnail" should be edited in `_config.yml` but these fields are actually in `_gallery_info.yml`.

### PR DESCRIPTION
**Pull request by brian-rose**
This PR get the Cookbook guide up to date and adds substantial detail on working with the Pythia infrastructure.

Closes #433

### CODE REVIEW

**Comment by brian-rose** [comment]
Here's a direct link to the preview of the new guide, since it's a bit hard to get to from the regular preview link: https://projectpythia.org/_preview/435/cookbook-guide.html

File: portal/cookbook-guide.md
Comment (clyne) [inline portal/cookbook-guide.md:98]:
```suggestion
In each of these files, in the field called `environment_name:`, replace  `cookbook-dev` with the name you used in your `environment.yml` file (probably `<your-cookbook-name>-dev`). Commit these changes.
```

File: portal/cookbook-guide.md
Comment (r-ford) [inline portal/cookbook-guide.md:154]:
```suggestion
- Create an up-to-date build environment using your `environment.yml` file
- Execute all the notebooks in that environment
- Package the book into a styled website
- Check all the links in the website to make sure they are valid
- Deploy the styled website so it's visible to the world
```

**Review by r-ford — APPROVED** [review]
I think all the changes here make sense.

This document is pretty large now, and I think it could be confusing for someone seeing it for the first time. I guess this is really a theme issue, but I can barely differentiate the heading levels, and there is no sidebar on the right with the section outline.

**Comment by brian-rose** [comment]
> I think all the changes here make sense.
> 
> This document is pretty large now, and I think it could be confusing for someone seeing it for the first time. I guess this is really a theme issue, but I can barely differentiate the heading levels, and there is no sidebar on the right with the section outline.

I had the same thought, I wished the level 2 and level 3 headings were easier to distinguish. I can try adding lines to separate the major sections.

Is the lack of a right sidebar a known theme issue? I see a button "toggle secondary sidebar" but it is non-functional.

**Comment by brian-rose** [comment]
Unfortunately horizontal lines don't help much because they are nearly invisible.

File: portal/cookbook-guide.md
Comment (clyne) [inline portal/cookbook-guide.md:37]: Small nit, but the option to "include all branches" occurs above the repository name selection

File: portal/cookbook-guide.md
Comment (clyne) [inline portal/cookbook-guide.md:38]:
```suggestion
1. Under Settings &rarr; Pages, ensure that GitHub Pages is enabled by checking that `Branch` is set to "gh-pages", and the folder set to "gh-pages/(root)". If it is not, change the Branch from "None" to "gh-pages/(root)" and click "Save"
```

File: portal/cookbook-guide.md
Comment (clyne) [inline portal/cookbook-guide.md:39]: The word "push" does not appear the page. The included image suggest that "Read and write permissions", under Workflow permissions", should be selected. Is this an additional step, or what is meant by allowing "push"?

File: portal/cookbook-guide.md
Comment (clyne) [inline portal/cookbook-guide.md:73]: There is another step following this one that asks which teams to give access to. Options are nobody, core, education, infrastructure, outreach, and "find a team". What is the guidance here?

**Review by clyne — COMMENTED** [review]
Review blocked until repository transfer issue is resolved

File: portal/cookbook-guide.md
Comment (brian-rose) [reply portal/cookbook-guide.md:73]: Hmm I guess the guidance should be to select "core". I think that includes all the active members of both "education" and "infrastructure". We haven't been very careful or consistent about how we use those teams.

Comment (brian-rose) [reply portal/cookbook-guide.md:73]: I selected "core" when I transferred a [new cookbook repo](https://github.com/ProjectPythia/radiative-feedback-cookbook) the other day, which is what got me started on this guide revision.

I forgot to include this guidance in the text.

File: portal/cookbook-guide.md
Comment (clyne) [inline portal/cookbook-guide.md:128]:
```suggestion
You should be able to build your new Cookbook on your local machine by running this command from the root of the repository:
```

File: portal/cookbook-guide.md
Comment (clyne) [inline portal/cookbook-guide.md:75]: After creating a personal fork of the repo I subsequently learned that GitHub had disabled all of the workflows and I had to re-enable them.

**Review by clyne — COMMENTED** [review]
Thanks for doing this, @brian-rose. It was badly needed

Following the instructions I was able to create a [test cookbook](https://github.com/ProjectPythia/hello-pythia-cookbook).  However, I did receive an error in one of the GitHub Actions as noted in my in-line comments. 

Also, a gh-pages website was never published, either in my personal repo or in the ProjectPythia repo for the cookbook. Not sure how to fix that.

**Comment by brian-rose** [comment]
> Also, a gh-pages website was never published, either in my personal repo or in the ProjectPythia repo for the cookbook. Not sure how to fix that.

It's here: https://projectpythia.org/hello-pythia-cookbook/README.html

File: portal/cookbook-guide.md
Comment (brian-rose) [reply portal/cookbook-guide.md:75]: > After creating a personal fork of the repo I subsequently learned that GitHub had disabled all of the workflows and I had to re-enable them.

This is expected! Normally we only want the workflows to operate on the upstream fork hosted in the ProjectPythia org. The typical and supported way to preview the changes that you make in your personal fork is to open a pull request on the upstream fork.

I can add this as a clarification in the guide.

There *may* be use cases where it makes sense to enable the workflows in the personal fork, but honestly I have never needed this for own cookbook development work.

**Comment by brian-rose** [comment]
> Following the instructions I was able to create a [test cookbook](https://github.com/ProjectPythia/hello-pythia-cookbook). However, I did receive an error in one of the GitHub Actions as noted in my in-line comments.

From what I can see [here](https://github.com/ProjectPythia/hello-pythia-cookbook/actions), all the actions are working as expected in the upstream fork.

File: portal/cookbook-guide.md
Comment (brian-rose) [reply portal/cookbook-guide.md:39]: I'll clarify the text. We mean that "Read and write permissions" should be selected. No additional steps.

File: portal/cookbook-guide.md
Comment (brian-rose) [reply portal/cookbook-guide.md]: Resolved: https://projectpythia.org/_preview/435/cookbook-guide.html#use-the-template

File: portal/cookbook-guide.md
Comment (brian-rose) [reply portal/cookbook-guide.md]: Resolved: https://projectpythia.org/_preview/435/cookbook-guide.html#use-the-template

File: portal/cookbook-guide.md
Comment (brian-rose) [reply portal/cookbook-guide.md:68]: Resolved: https://projectpythia.org/_preview/435/cookbook-guide.html#steps-to-transfer-the-repository

**Comment by brian-rose** [comment]
@clyne thanks for the detailed read-through! I think I have addressed all your comments, and I added a bit more detail on making Pull Requests.

In retrospect, I think it might be better to separate this guide into two pieces:
- contributing to Cookbooks
- making new Cookbooks

But that could be left as an open issue for the future if we're reasonably happy with what's here.

File: portal/cookbook-guide.md
Comment (clyne) [reply portal/cookbook-guide.md:73]: Gotcha. Clarifying would help.

**Comment by clyne** [comment]
> > Also, a gh-pages website was never published, either in my personal repo or in the ProjectPythia repo for the cookbook. Not sure how to fix that.
> 
> It's here: https://projectpythia.org/hello-pythia-cookbook/README.html

Sorry, should have been more clear. The gh-pages site is not linked from underneath the "About" heading on the repo [website](https://github.com/ProjectPythia/hello-pythia-cookbook). Shouldn't this happen automatically?

File: portal/cookbook-guide.md
Comment (clyne) [inline portal/cookbook-guide.md:149]: So "your Cookbook repo" is a little misleading here? The actions are only triggered on the cookbook that is in the ProjectPythia organization, not your fork of the repo, correct?

**Review by clyne — COMMENTED** [review]
Just a couple more minor in-line comments.

**Comment by brian-rose** [comment]
> > > Also, a gh-pages website was never published, either in my personal repo or in the ProjectPythia repo for the cookbook. Not sure how to fix that.
> > 
> > 
> > It's here: https://projectpythia.org/hello-pythia-cookbook/README.html
> 
> Sorry, should have been more clear. The gh-pages site is not linked from underneath the "About" heading on the repo [website](https://github.com/ProjectPythia/hello-pythia-cookbook). Shouldn't this happen automatically?

No, it does not happen automatically. Maybe you missed my suggestion for this in [this section](https://projectpythia.org/_preview/435/cookbook-guide.html#where-is-my-book) under "Here's a handy trick..."

File: portal/cookbook-guide.md
Comment (brian-rose) [reply portal/cookbook-guide.md:149]: I added a note block at the bottom of [this section](https://projectpythia.org/_preview/435/cookbook-guide.html#where-is-my-book) to try to clarify this. 

We can use more precise language in the sentence you highlighted. I didn't want it to get too convoluted. Strictly speaking, it's the `main` branch of the `upstream` repository, not your personal fork. Unless you *haven't* transferred ownership of your repo, it which case it would be the `main` branch of your personal repository (which is not a fork and thus does not have an upstream equivalent).

Comment (brian-rose) [reply portal/cookbook-guide.md:149]: Any suggested language for this?

File: portal/cookbook-guide.md
Comment (brian-rose) [reply portal/cookbook-guide.md:73]: Clarification is now in the note block at the end of [this section](https://projectpythia.org/_preview/435/cookbook-guide.html#where-is-my-book).

File: portal/cookbook-guide.md
Comment (clyne) [reply portal/cookbook-guide.md:149]: Nope. That looks good.

**Review by clyne — APPROVED** [review]

**Comment by brian-rose** [comment]
> > > > > Also, a gh-pages website was never published, either in my personal repo or in the ProjectPythia repo for the cookbook. Not sure how to fix that.
> > > > 
> > > > 
> > > > It's here: https://projectpythia.org/hello-pythia-cookbook/README.html
> > > 
> > > 
> > > Sorry, should have been more clear. The gh-pages site is not linked from underneath the "About" heading on the repo [website](https://github.com/ProjectPythia/hello-pythia-cookbook). Shouldn't this happen automatically?
> > 
> > 
> > No, it does not happen automatically. Maybe you missed my suggestion for this in [this section](https://projectpythia.org/_preview/435/cookbook-guide.html#where-is-my-book) under "Here's a handy trick..."
> 
> No, I didn't miss it, but it explains the rabbit hole that I went down. The "About" settings are different for my personal repo, and do not include the option to "Use your GitHub Pages website". I probably got confused about which repo I was looking at. Here's a screenshot for what I see on my personal repo:
> 
> About

Oh, that makes sense! I will add a clarifying note. It's easy to get confused about which version of the repo one is looking at.

**Comment by clyne** [comment]
> > > > > > Also, a gh-pages website was never published, either in my personal repo or in the ProjectPythia repo for the cookbook. Not sure how to fix that.
> > > > > 
> > > > > 
> > > > > It's here: https://projectpythia.org/hello-pythia-cookbook/README.html
> > > > 
> > > > 
> > > > Sorry, should have been more clear. The gh-pages site is not linked from underneath the "About" heading on the repo [website](https://github.com/ProjectPythia/hello-pythia-cookbook). Shouldn't this happen automatically?
> > > 
> > > 
> > > No, it does not happen automatically. Maybe you missed my suggestion for this in [this section](https://projectpythia.org/_preview/435/cookbook-guide.html#where-is-my-book) under "Here's a handy trick..."
> > 
> > 
> > No, I didn't miss it, but it explains the rabbit hole that I went down. The "About" settings are different for my personal repo, and do not include the option to "Use your GitHub Pages website". I probably got confused about which repo I was looking at. Here's a screenshot for what I see on my personal repo:
> > About
> 
> Oh, that makes sense! I will add a clarifying note. It's easy to get confused about which version of the repo one is looking at.

Cool. I'll be going offline soon and have approved the PR. Nice work!

File: portal/cookbook-guide.md
Comment (brian-rose) [inline portal/cookbook-guide.md:166]: @clyne I added one last clarification here

Comment (brian-rose) [reply portal/cookbook-guide.md:166]: I should have merged the PR while I had approval, lol. If you have a chance to approve again, that would be terrific.

**Review by clyne — APPROVED** [review]

File: portal/cookbook-guide.md
Comment (clyne) [reply portal/cookbook-guide.md:166]: Done!

**Comment by brian-rose** [comment]
Thanks so much @clyne for the careful review! Much appreciated.

**Pull request merged**
